src/devices/rojaflex.c

```diff
@@ -0,0 +1,317 @@
+    }
+    else {
+        return DECODE_ABORT_EARLY; // no preamble detected
+    }
+
+    if (dataframe_bitcount < (DATAFRAME_BITCOUNT_INCL_CRC - MESSAGE_CRC_BITCOUNT) || (dataframe_bitcount > DATAFRAME_BITCOUNT_INCL_CRC)) {
+        // check min and max length
+        return DECODE_ABORT_LENGTH;
+    }
+
+    //Extract raw line
+    bitbuffer_extract_bytes(bitbuffer, row, start_pos + sizeof(message_preamble) * 8, msg, dataframe_bitcount);
+    if (decoder->verbose > 1) {
+        bitrow_printf(msg, dataframe_bitcount, "%s: frame data: ", __func__);
+    }
+
+    // Check CRC if available
+    if (dataframe_bitcount == DATAFRAME_BITCOUNT_INCL_CRC) {
+        // Thanks to: ./reveng -w 16 -s $msg1 $msg2 $msg3
+        // width=16  poly=0x8005  init=0xffff  refin=false  refout=false  xorout=0x0000  check=0xaee7  residue=0x0000  name="CRC-16/CMS"
+        uint16_t crc_message = (msg[MESSAGE_CRC_OFFSET] << 8 | msg[MESSAGE_CRC_OFFSET + 1]);
+        uint16_t crc_calc    = crc16(&msg[LENGTH_OFFSET], 9, 0x8005, 0xffff);
+
+        if (crc_message != crc_calc) {
+            if (decoder->verbose)
+                fprintf(stderr, "%s: CRC invalid message:%04x != calc:%04x\n", __func__, crc_message, crc_calc);
+
+            return DECODE_FAIL_MIC;
+        };
+    }
+
+    // Build the default terminal output
+    {
+        char tokenString[7] = "";
+        char id[10]         = "";
+        char *deviceType    = "unknown";
+        sprintf(tokenString, "0x%02x%02x", msg[MESSAGE_TOKEN_OFFSET], msg[MESSAGE_TOKEN_OFFSET + 1]);
+        sprintf(id, "0x%02x%02x%02x%01x", msg[ID_OFFSET], msg[ID_OFFSET + 1], msg[ID_OFFSET + 2], (msg[ID_OFFSET + 3] >> 4));
+
+        if ((msg[COMMAND_ID_OFFSET] & 0xF) == 0x5) {
+            deviceType = "RojaFlex-Shutter";
+        }
+        else if ((msg[COMMAND_ID_OFFSET] & 0xF) == 0xa) {
+            // Rojaflex Bridge clones a remote signal but does not send an CRC!?!?
+            // So we can detect if it a real remote or a bridge on the message length.
+            if (dataframe_bitcount == DATAFRAME_BITCOUNT_INCL_CRC) {
+                deviceType = "RojaFlex-Remote";
+            }
+            else {
+                deviceType = "RojaFlex-Bridge";
+            }
+        }
+
+        /* clang-format off */
+        data = data_make(
+                    "model",        "Model",     DATA_STRING, deviceType,
+                    "id",           "ID",        DATA_STRING, id,
+                    "channel",      "Channel",   DATA_INT,    msg[CHANNEL_OFFSET] & 0xF,
+                    "token",        "Msg Token", DATA_STRING, tokenString,
+                    "commandtype",  "Command ",  DATA_STRING, getCommandString(&msg[0]),
+                    "commandvalue", "Value",     DATA_INT,    msg[COMMAND_VALUE_OFFSET],
+                    "mic",          "Integrity", DATA_STRING, (dataframe_bitcount == DATAFRAME_BITCOUNT_INCL_CRC) ? "CRC" : "",
```

Comment:
Only include mic if there is a message integrity check. Add it conditionally.

Look here how to do it:
https://github.com/merbanan/rtl_433/blob/master/src/devices/ambientweather_tx8300.c#L114

Reply:
Done